vertical-pod-autoscaler/pkg/utils/vpa/api.go

```diff
@@ -162,21 +143,46 @@ func GetControllingVPAForPod(pod *core.Pod, vpas []*VpaWithSelector, ctrlFetcher
 	// Choose the strongest VPA from the ones that match this Pod.
 	for _, vpaWithSelector := range vpas {
 		if vpaWithSelector.Vpa.Spec.TargetRef == nil {
+			klog.V(5).Infof("skipping VPA object %s because targetRef is not defined. If this is a v1beta1 object, switch to v1", klog.KObj(vpaWithSelector.Vpa))
 			continue
 		}
 		if vpaWithSelector.Vpa.Spec.TargetRef.Kind != parentController.Kind ||
 			vpaWithSelector.Vpa.Namespace != parentController.Namespace ||
 			vpaWithSelector.Vpa.Spec.TargetRef.Name != parentController.Name {
 			continue // This pod is not associated to the right controller
 		}
-		if PodMatchesVPA(pod, vpaWithSelector) && stronger(vpaWithSelector.Vpa, controllingVpa) {
+		if PodMatchesVPA(pod, vpaWithSelector) && Stronger(vpaWithSelector.Vpa, controllingVpa) {
 			controlling = vpaWithSelector
 			controllingVpa = controlling.Vpa
 		}
 	}
 	return controlling
 }
 
+// FindParentControllerForPod returns the parent controller (topmost well-known or scalable controller) for the given Pod.
+func FindParentControllerForPod(pod *core.Pod, ctrlFetcher controllerfetcher.ControllerFetcher) (*controllerfetcher.ControllerKeyWithAPIVersion, error) {
```

Comment:
Can a test be written for this function?

Reply:
I added unit test in https://github.com/kubernetes/autoscaler/pull/7010/commits/dcbd7164678527fb9ed867c20c4a74215d026c38.